src/IIIFPresentation/API/Features/Storage/StorageController.cs

```diff
@@ -93,49 +97,66 @@ public async Task<IActionResult> Get(int customerId, string id, int? page = 1, i
     [Authorize]
     [HttpPost("collections")]
     [ETagCaching]
-    public async Task<IActionResult> Post(int customerId, [FromServices] UpsertFlatCollectionValidator validator)
+    public async Task<IActionResult> Post(int customerId, [FromServices] PresentationCollectionValidator validator)
     {
         if (!Request.HasShowExtraHeader()) return this.Forbidden();
         
         var rawRequestBody = await Request.GetRawRequestBodyAsync();
         
-        var collection = JsonConvert.DeserializeObject<UpsertFlatCollection>(rawRequestBody);
+        var deserializedCollection = await TryDeserializePresentationCollection(rawRequestBody);
+        if (deserializedCollection.Error)  return PresentationUnableToSerialize();
 
-        if (collection == null)
-        {
-            return this.PresentationProblem("Could not deserialize collection", null, (int)HttpStatusCode.BadRequest,
-                "Deserialization Error");
-        }
-
-        var validation = await validator.ValidateAsync(collection);
+        var validation = await validator.ValidateAsync(deserializedCollection.ConvertedIIIF);
         
         if (!validation.IsValid)
         {
             return this.ValidationFailed(validation);
         }
         
-        return await HandleUpsert(new CreateCollection(customerId, collection!, rawRequestBody, GetUrlRoots()));
+        return await HandleUpsert(new CreateCollection(customerId, deserializedCollection.ConvertedIIIF, rawRequestBody, GetUrlRoots()));
     }
     
     [Authorize]
     [HttpPut("collections/{id}")]
     [ETagCaching]
-    public async Task<IActionResult> Put(int customerId, string id, [FromBody] UpsertFlatCollection collection, 
-        [FromServices] UpsertFlatCollectionValidator validator)
+    public async Task<IActionResult> Put(int customerId, string id, 
```

Comment:
With the exception of the `IRequest` type it looks like `Put` and `Post` are identical and a candidate for refactoring